pkg/certificate/providers/mrc.go

```diff
@@ -6,15 +6,15 @@ import (
 	"k8s.io/client-go/tools/cache"
 
 	"github.com/openservicemesh/osm/pkg/apis/config/v1alpha2"
+	"github.com/openservicemesh/osm/pkg/catalog"
 	"github.com/openservicemesh/osm/pkg/certificate"
-	"github.com/openservicemesh/osm/pkg/compute"
 )
 
 // MRCComposer is a composer object that allows consumers
 // to observe MRCs (via List() and Watch()) as well as generate
 // `certificate.Provider`s from those MRCs
 type MRCComposer struct {
-	compute.Interface
+	catalog.Interface
```

Comment:
"interfaces should be defined where they are consumed"

let's create a new interface here with *just* the methods required.

Reply:
I believe GetMeshConfig() is the only method used by pkg/certificate/providers -- does this warrant a whole new interface?

Reply:
yup. It keeps packages isolated and decoupled from each other. It allows allows making very trivial implementations for testing purposes if desired.